caput/memh5.py

```diff
@@ -1282,9 +1450,11 @@ class MemDiskGroup(_BaseGroup):
     detect_subclass: boolean, optional
         If *data_group* is specified, whether to inspect for a
         '__memh5_subclass' attribute which specifies a subclass to return.
+    file_format : `fileformats.FileFormat`
+        File format to use. File format will be guessed if not supplied. Default `None`.
     """
 
-    def __init__(self, data_group=None, distributed=False, comm=None):
+    def __init__(self, data_group=None, distributed=False, comm=None, file_format=None):
```

Comment:
I think `file_format` can have a fallback value of `None` in a couple of places, and I am wondering if this ever breaks anything. Testing a form of `.yaml` where you have a `.zarr` filename, but do not specify the `output_format`.

Reply:
You can remove the `output_format` and `.zarr` outputs are still accurately typed. Guessing this is fine, then.